### ISSUE TITLE
Kennards Hire (AU/NZ)

### ISSUE THREAD

**Issue by davidhicks** [alltheplaces/alltheplaces#6544]
### Brand name

Kennards Hire

### Wikidata ID

Q63557330

### Store finder url(s)

https://www.kennards.com.au/sitemap.xml and https://www.kennardshire.co.nz/sitemap.xml

(filter for https://www.kennards.com.au/branch/ and https://www.kennardshire.co.nz/branch/)

**Comment by CloCkWeRX** [comment]
This one seems a bit tricky, looking at https://www.kennards.com.au/branch/abbotsford they aren't providing a lat/lon.

The rendering of addresses etc isn't particularly fun to parse either.

**Comment by Cj-Malone** [comment]
It looks like there is a JSON blob with all the good stuff, `//script[@id="__NEXT_DATA__"]/text()`.

Take a look at the Harvey Norman spider, it should be pretty similar.

**Comment by CloCkWeRX** [comment]
https://github.com/search?q=repo%3Aalltheplaces%2Falltheplaces%20NEXT_DATA&type=code there's... actually quite a lot of these. Do we know what the name for this is?

**Comment by Cj-Malone** [comment]
I've always assumed it's from next.js. Not checked though. But it's Json not JavaScript, so it's easy enough to parse.

**Comment by davidhicks** [comment]
> I've always assumed it's from next.js. Not checked though. But it's Json not JavaScript, so it's easy enough to parse.

Yep, it's just a pattern common to React web applications. The data itself can wildly change depending on the brand's store locator. Reference: https://github.com/vercel/next.js/discussions/15117
